form field error message - html bepaald - acceptance criteria (#1103)
nl-design-system/candidate: packages/docs/form-field-error-message-docs/docs/acceptance-criteria.md

<!-- @license CC0-1.0 -->

# Acceptatiecriteria voor het maken van de Form Field Error Message component

## Semantiek

Dit zijn de acceptatiecriteria welke HTML elementen en HTML attributen het beste zijn voor deze component.

### Welke HTML elementen het beste zijn voor deze component

#### Must Have elementen

- De Form Field Error Message is opgemaakt met het `div` element.

> - De Form Field Error Message `div` element bevat een `div` element voor de content.

Reviewer: ```suggestion
- De Form Field Error Message `div` element bevat een `div` element voor de Content.
```

Author: Heb je verwerkt in https://github.com/nl-design-system/candidate/pull/1264 dus ik sluit deze comment (ben de twee duplicate PRs aan het vergelijken en resolven)

- De Form Field Error Message actions komt na de content.

#### Could Have elementen

- De Form Field Error Message `div` element bevat een `div` element voor de actions.

> - De Form Field Error Message content mag een inhoud bevatten welke bestaat in de vorm van een `p` element en [phrasing content](https://www.w3.org/TR/2011/WD-html5-20110525/content-models.html#phrasing-content-0) met uitzondering van [interactive content](https://www.w3.org/TR/2011/WD-html5-20110525/content-models.html#interactive-content-0).

Reviewer: Ik denk dat je technically ook wel een beknopte `ul` kan doen als je de tekst goed aanpakt

Author: Deze heb je in de nieuwe PR https://github.com/nl-design-system/candidate/pull/1264 veranderd naar dit:
- De Form Field Error Message Content mag een inhoud bevatten welke bestaat in de vorm van een `p` HTML-element en [phrasing content](https://www.w3.org/TR/2011/WD-html5-20110525/content-models.html#phrasing-content-0) met uitzondering van [interactive content](https://www.w3.org/TR/2011/WD-html5-20110525/content-models.html#interactive-content-0).

> - De Form Field Error Message actions mag een inhoud bevatten welke bestaat in de vorm van een `p` element en [phrasing content](https://www.w3.org/TR/2011/WD-html5-20110525/content-models.html#phrasing-content-0).

Reviewer: Er mag toch ook een button in enzo? Waarom is de `p` element zo belangrijk? Voelt alsof we hier gewoon een brede HTML-categorie kunnen noemen

Author: Heb je verwerkt in https://github.com/nl-design-system/candidate/pull/1264 dus ik sluit deze comment (ben de twee duplicate PRs aan het vergelijken en resolven)

> - De Form Field Error Message actions mag een inhoud bevatten welke bestaat in de vorm van een `p` element en [phrasing content](https://www.w3.org/TR/2011/WD-html5-20110525/content-models.html#phrasing-content-0).

Reviewer: Was het nou zo dat ik had gehallicuneerd dat Form Field Error Message "Actions" kreeg en dat die eigenlijk alleen bij Form Field Status mochten?

Author: We hadden eerst besloten: krijgt wel Actions, als learning uit Alert. Daarna besloten: nee, toch niet, die moeten er uit, en pas wanneer hij in een Form Field Status word gebruikt. Inderdaad verwijderd in de nieuwe PR https://github.com/nl-design-system/candidate/pull/1264. Ik sluit deze comment

### Welke HTML attributen het beste zijn voor deze component

#### Must Have attributen

- De Form Field Error Message content bevat altijd een `id` attribuut, zodat het met `aria-describedby` attribuut aan het input-element gekoppeld kan worden.

> - De Form Field Error Message gebruikt geen `tabindex` attribuut, de component is ondersteunend, niet interactief.

Reviewer: ```suggestion
- De Form Field Error Message gebruikt geen `tabindex` attribuut, de component is informatief, niet interactief.
```

Author: Verwerkt in nieuwe PR https://github.com/nl-design-system/candidate/pull/1264

- De Form Field Error Message inhoud is alleen zichtbaar en toegankelijk wanneer er een fout is.
- Wanneer de tekst in realtime wordt ingevuld, is de Form Field Error Message altijd aanwezig in de DOM en in de accessibility tree.

#### Could Have attributen

> - Wanneer de tekst in realtime wordt ingevuld, maakt de Form Field Error Message content gebruik van `role="alert"` attribuut.

Reviewer: ```suggestion
- In zeldzame situaties waar een statusbericht nodig is, kan de Form Field Error Message gebruik maken van `role="alert"` attribuut.
```

Author: Deze hebben we in de review verwijderd in de nieuwe PR https://github.com/nl-design-system/candidate/pull/1264
